tests/sentry/web/frontend/test_auth_mobile.py

```diff
@@ -0,0 +1,209 @@
+from urllib.parse import parse_qs, urlparse
+
+from django.test import TestCase
+from django.urls import reverse
+
+from sentry.testutils.cases import TestCase as SentryTestCase
+from sentry.utils.auth import is_valid_redirect
+
+
+class MobileAuthenticationTest(SentryTestCase):
+    """
+    Test cases for mobile authentication flow using custom URL schemes.
+    """
+
+    def setUp(self):
+        super().setUp()
+        self.organization = self.create_organization(name="Test Org", slug="test-org")
+        self.user = self.create_user("test@example.com")
+
+    def test_is_valid_redirect_with_mobile_scheme(self):
+        """Test that mobile custom URL schemes are considered valid redirects."""
+        # Test valid mobile scheme
+        self.assertTrue(is_valid_redirect("sentry-mobile-agent://auth"))
+        self.assertTrue(is_valid_redirect("sentry-mobile-agent://auth/callback"))
+
+        # Test invalid schemes
+        self.assertFalse(is_valid_redirect("malicious-scheme://evil.com"))
+        self.assertFalse(is_valid_redirect("http://evil.com"))
+        self.assertFalse(is_valid_redirect(""))
+        self.assertFalse(is_valid_redirect(None))
+
+    def test_organization_login_with_mobile_next_parameter(self):
+        """Test that the organization login page accepts mobile next parameter."""
+        url = reverse("sentry-auth-organization", args=[self.organization.slug])
+        mobile_redirect = "sentry-mobile-agent://auth"
+
+        response = self.client.get(url, {"next": mobile_redirect})
+
+        # Should show login page without redirect error
+        self.assertEqual(response.status_code, 200)
+
+        # Check that the next URL is preserved in session
+        session = self.client.session
+        self.assertEqual(session.get("_next"), mobile_redirect)
+
+    def test_successful_login_redirects_to_mobile_app(self):
+        """Test that successful authentication redirects to mobile app."""
+        url = reverse("sentry-auth-organization", args=[self.organization.slug])
+        mobile_redirect = "sentry-mobile-agent://auth"
+
+        # Login the user first
+        self.login_as(self.user)
+
+        # Make request with mobile redirect parameter as authenticated user
+        response = self.client.get(url, {"next": mobile_redirect})
+
+        # Should redirect to the mobile app since user is already authenticated
+        self.assertEqual(response.status_code, 302)
+        # The redirect location should be the mobile custom URL scheme
+        self.assertEqual(response["Location"], mobile_redirect)
+
+
+class MobileAuthValidationTest(TestCase):
+    """
+    Unit tests for mobile authentication validation functions.
```

Comment:
Nit: I personally don't find these comments super helpful, would recommend removing them for brevity in the file